vertx-mysql-client/src/main/java/io/vertx/mysqlclient/impl/codec/InitCommandCodec.java

```diff
@@ -175,12 +202,52 @@ private void decodeInit1(InitCommand cmd, ByteBuf payload) {
       case ERROR_PACKET_HEADER:
         handleErrorPacketPayload(payload);
         break;
+      case AUTH_SWITCH_REQUEST_STATUS_FLAG:
+        // Protocol::AuthSwitchRequest
+        payload.skipBytes(1); // status flag, always 0xFE
+        String pluginName = BufferUtils.readNullTerminatedString(payload, StandardCharsets.UTF_8);
+        switch (pluginName) {
+          case "mysql_native_password":
+            byte[] challenge = new byte[20];
+            payload.readBytes(challenge);
+            byte[] scrambledPassword = Native41Authenticator.encode(cmd.password(), StandardCharsets.UTF_8, challenge);
+
+            ByteBuf packet = allocateBuffer(24);
+            packet.writeMediumLE(20);
+            packet.writeByte(sequenceId);
+            packet.writeBytes(scrambledPassword);
+            sendNonSplitPacket(packet);
+
+            break;
+          default:
+            //TODO support more auth methods here
+            completionHandler.handle(CommandResponse.failure(new UnsupportedOperationException("Unsupported authentication method: " + pluginName)));
+            return;
+        }
+        break;
       default:
         throw new UnsupportedOperationException();
     }
   }
 
-  private void sendHandshakeResponseMessage(String username, String password, String database, int collationId, byte[] serverScramble, String authMethodName, Map<String, String> clientConnectionAttributes) {
+  private void sendSslRequest() {
+    ByteBuf packet = allocateBuffer(36);
+    // encode packet header
+    packet.writeMediumLE(32);
+    packet.writeByte(sequenceId);
+
+    // encode SSLRequest payload
+    packet.writeIntLE(encoder.clientCapabilitiesFlag);
+    packet.writeIntLE(PACKET_PAYLOAD_LENGTH_LIMIT);
+    checkCollation();
+    packet.writeByte(collation.collationId());
+    byte[] filler = new byte[23];
+    packet.writeBytes(filler);
```

Comment:
use `ByteBuf#writeZero(int)` rather than creating a temporary `byte[]` and writing int.

Reply:
updated in https://github.com/eclipse-vertx/vertx-sql-client/pull/411/commits/fb664fa985acf5a8281f760b2a0689567ada3a2e.